test/app_test.dart

```diff
@@ -292,6 +292,32 @@ Future<void> main([List<String>? args]) async {
     expect(receivedPerson1.name, arg1.name);
     expect(receivedPerson2.name, arg2.name);
   });
+
+  baasTest('App.reconnect', (appConfiguration) async {
+    final app = App(appConfiguration);
+
+    final user = await app.logIn(Credentials.anonymous());
+    final configuration = Configuration.flexibleSync(user, [Task.schema]);
+    final realm = getRealm(configuration);
+    final session = realm.syncSession;
+
+    // TODO: We miss a way to force a disconnect. Once we implement GenericNetworkTransport
```

Comment:
can we use `HttpClient.close()`?

Reply:
Problem is that core does not use the http transport abstraction during sync, only during authentication and such.. so we need a few more bits in place first

Reply:
I believe we can (and should) customize this by creating our own network transport, but it is perhaps a bit beyond this particular task.